### PR TITLE
Creating beaver pid directory when missing

### PR DESCRIPTION
**Pull request by BjoernT**
#564 Fixing Beaver PID handling
- Creating Beaver PID directory when missing
- Cleaning Beaver PID after shutting it down

### CODE REVIEW

**Comment by BjoernT** [comment]
Updated beaver_pre_install.yml to remove the pid dir

File: rpcd/playbooks/roles/beaver/templates/beaver.j2 [hunk 1]
```diff
@@ -37,6 +37,7 @@ do_start() {
 }
 
 do_stop() {
+    test -d "$( dirname $BEAVER_PID )" || mkdir "$( dirname $BEAVER_PID )" && chown $BEAVER_NAME:$BEAVER_NAME "$( dirname $BEAVER_PID )"
```
Comment (prometheanfire) [inline rpcd/playbooks/roles/beaver/templates/beaver.j2]: shouldn't this be in start, not stop?

also, doing it all on one line is generally bad practice, can you fix that?

    local BEAVER_PID_DIRNAME="$(dirname $BEAVER_PID)"
    if [[ ! -d $BEAVER_PID_DIRNAME ]]; then
        mkdir -p "${BEAVER_PID_DIRNAME}"
        chown "${BEAVER_NAME}":"${BEAVER_NAME}"
    fi

**Comment by BjoernT** [comment]
Oh yes you're right, cut'n paste victim... let me change it

**Comment by prometheanfire** [comment]
can you make that a proper conditional?

**Comment by BjoernT** [comment]
Yeah "proper" is subjective I personally like on liners... but I'm fine with that

File: rpcd/playbooks/roles/beaver/templates/beaver.j2 [hunk 2]
```diff
@@ -60,6 +67,7 @@ do_stop() {
         I="$((I+1))"
     done
     echo '.'
+    test -f ${BEAVER_PID} && rm -f ${BEAVER_PID}
```
Comment (prometheanfire) [inline rpcd/playbooks/roles/beaver/templates/beaver.j2]: this seems new, why is it needed?

Comment (BjoernT) [inline rpcd/playbooks/roles/beaver/templates/beaver.j2]: Because beaver is not deleting the file after it stopped which is quite annoying.

Comment (prometheanfire) [inline rpcd/playbooks/roles/beaver/templates/beaver.j2]: ah, this is part of stop, didn't see that at first

**Comment by prometheanfire** [comment]
:+1:

File: rpcd/playbooks/roles/beaver/templates/beaver.j2 (hunk 2 above)
Comment (BjoernT) [inline rpcd/playbooks/roles/beaver/templates/beaver.j2]: No worries, I'll update the commit message

Comment (git-harry) [inline rpcd/playbooks/roles/beaver/templates/beaver.j2]: @BjoernT don't forget to update the commit message.

**Comment by d34dh0r53** [comment]
@BjoernT update the commit message, rebase and :+1:

**Comment by BjoernT** [comment]
Yeah I had it re based this morning and updated the message already. Re based again.

**Pull request merged**